backend/inspirehep/records/tasks.py

```diff
@@ -33,6 +40,12 @@ def update_records_relations(uuids):
                     record.update_conference_paper_and_proccedings()
                     record.update_institution_relations()
                     record.update_experiment_relations()
+                deleted_record_uri = record.check_if_record_was_deleted_by_merge()
+                if deleted_record_uri:
+                    record_ref = get_value(record, "self.$ref")
+                    recalculate_references_for_merged_record.delay(
+                        deleted_record_uri, record_ref
+                    )
```

Comment:
you shouldn't add this here:
* it has nothing to do with this function, which updates relations inside auxiliary database tables
* it has a race condition, because the task might be executed before the changes have been persisted to the database

I understand why you did this, because you can't easily perform the check in the `models_committed` receiver as the SQLAlchemy session is in a state that doesn't allow you to execute any DB query, which would be needed to get the previous version. But it's definitely not the right place. I don't know where to put this, maybe in the task that does the reindexing (ask the other devs).